File: brasilburger-java/src/main/java/sn/ism/brasilburger/model/Complement.java
package sn.ism.brasilburger.model;

import java.math.BigDecimal;

public class Complement {
    private int id;
    private String nom;
    private TypeComplement type;
    private BigDecimal prix;
    private String image;
    private boolean archived;

    public Complement() {
    }

    public Complement(int id, String nom, TypeComplement type, BigDecimal prix, String image, boolean archived) {
        this.id = id;
        this.nom = nom;
        this.type = type;
        this.prix = prix;
        this.image = image;
        this.archived = archived;
    }

    public Complement(String nom, TypeComplement type, BigDecimal prix, String image) {
        this(0, nom, type, prix, image, false);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public TypeComplement getType() {
        return type;
    }

    public void setType(TypeComplement type) {
        this.type = type;
    }

    public BigDecimal getPrix() {
        return prix;
    }

    public void setPrix(BigDecimal prix) {
        this.prix = prix;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public boolean isArchived() {
        return archived;
    }

    public void setArchived(boolean archived) {
        this.archived = archived;
    }

    @Override
    public String toString() {
        return "Complement{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", type=" + type +
                ", prix=" + prix +
                ", image='" + image + '\'' +
                ", archived=" + archived +
                '}';
    }
}

File: src/Entity/Complement.php
<?php

namespace App\Entity;

use App\Enum\TypeComplementEnum;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'complement')]
class Complement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_complement', type: 'integer')]
    private ?int $idComplement = null;

    #[ORM\Column(type: 'string', length: 120)]
    private string $nom;

    #[ORM\Column(enumType: TypeComplementEnum::class)]
    private TypeComplementEnum $typeComplement;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private string $prix;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'boolean', options: ['default' => false])]
    private bool $isArchived = false;

    public function getIdComplement(): ?int { return $this->idComplement; }
    public function getNom(): string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    public function getTypeComplement(): TypeComplementEnum { return $this->typeComplement; }
    public function setTypeComplement(TypeComplementEnum $t): self { $this->typeComplement = $t; return $this; }
    public function getPrix(): string { return $this->prix; }
    public function setPrix(string $prix): self { $this->prix = $prix; return $this; }
    public function getImage(): ?string { return $this->image; }
    public function setImage(?string $image): self { $this->image = $image; return $this; }
    public function isArchived(): bool { return $this->isArchived; }
    public function setIsArchived(bool $v): self { $this->isArchived = $v; return $this; }
}
